### ISSUE TITLE
Fahrenheit option for weather

### ISSUE THREAD

**Issue by VmKid** [Freeyourgadget/Gadgetbridge#858]
(Edited for slight correction)

This was brought up briefly in #467 but quite a while ago.

Currently (0.22.0) the weather will only display in Celsius. A toggle format setting would be greatly appreciated.

**Comment by danielegobbetti** [comment]
You need to enable the *experimental* background javascript support and configure the watchface to use fahrenheit, then it will work.

Please note that you have to disconnect / connect again after enabling the background javascript execution. And also note that the unit change will be visible (depending on the watchface) only after fresh weather data are fetched from the weather app.

**Comment by VmKid** [comment]
Thanks for the workaround, however I'm now having an issue with the JS support option. After enabling it, Gadgetbridge crashes upon trying to connect to the device. The log doesn't appear to have any exception information, mostly just regular debug output:

```
04:01:12.323 [main] INFO  n.f.g.Logging - Gadgetbridge version: 0.22.0
04:01:12.487 [main] INFO  n.f.g.u.GB - Loading apps_blacklist
04:01:12.487 [main] INFO  n.f.g.u.GB - Loaded apps_blacklist has 1 entries
04:01:12.487 [main] INFO  n.f.g.u.GB - Loading calendars_blacklist
04:01:12.488 [main] INFO  n.f.g.u.GB - Loaded calendars_blacklist has 6 entries
04:01:12.718 [main] INFO  n.f.g.s.NotificationCollectorMonitorService - ensureCollectorRunning collectorComponent: ComponentInfo{nodomain.freeyourgadget.gadgetbridge/nodomain.freeyourgadget.gadgetbridge.externalevents.NotificationListener}
04:01:12.727 [main] DEBUG n.f.g.s.NotificationCollectorMonitorService - ensureCollectorRunning: collector not running, reviving...
04:01:12.727 [main] DEBUG n.f.g.s.NotificationCollectorMonitorService - toggleNotificationListenerService() called
04:01:12.746 [main] DEBUG n.f.g.s.DeviceCommunicationService - DeviceCommunicationService is being created
04:01:12.747 [main] DEBUG n.f.g.s.DeviceCommunicationService - Service startcommand: nodomain.freeyourgadget.gadgetbridge.devices.action.start
04:01:12.757 [main] DEBUG n.f.g.s.DeviceCommunicationService - Service startcommand: nodomain.freeyourgadget.gadgetbridge.devices.action.request_deviceinfo
```

**Comment by danielegobbetti** [comment]
Please try to force close GB and start it again. The crash might be visible in the system-wide ``adb logcat`` output, as it's possibly chrome that is crashing when the background webview gets created (and this happens when connecting), if you could take a look (search for "chrome" or " webview") I'd appreciate that.

If you disable background JS everything should work as usual, the error shouldn't be persistent.

I don't think this will change the system weather app temperature unit, at least nothing I am aware of is doing this explicitly, but let's hear what @ashimokawa has to say about this.

**Comment by VmKid** [comment]
Judging by the logs, it doesn't seem to be webview in my case. [Here](https://pastebin.com/4JhGRW29) I've pastebinned my logcat, only showing GB events. Of particular note is what appears to be the actual exception:

```
10-29 14:04:46.206 20413 14123 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: Uncaught exception: Calling startActivity() from outside of an Activity  context requires the FLAG_ACTIVITY_NEW_TASK flag. Is th...
10-29 14:04:46.206 20413 14123 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler:  at android.app.ContextImpl.startActivity(ContextImpl.java:738) ~[na:0.0]                                                          
10-29 14:04:46.206 20413 14123 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler:  at android.app.ContextImpl.startActivity(ContextImpl.java:725) ~[na:0.0]                                                          
10-29 14:04:46.206 20413 14123 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler:  at android.content.ContextWrapper.startActivity(ContextWrapper.java:345) ~[na:0.0]                                                
10-29 14:04:46.206 20413 14123 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler:  at nodomain.freeyourgadget.gadgetbridge.service.devices.pebble.PebbleIoThread.connect(PebbleIoThread.java:170) ~[na:0.0]          
10-29 14:04:46.206 20413 14123 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler:  at nodomain.freeyourgadget.gadgetbridge.service.devices.pebble.PebbleIoThread.run(PebbleIoThread.java:197) ~[na:0.0]              
10-29 14:04:46.206 20413 14123 E AndroidRuntime: Process: nodomain.freeyourgadget.gadgetbridge, PID: 20413                                                                                                                        
10-29 14:04:46.206 20413 14123 E AndroidRuntime:    at nodomain.freeyourgadget.gadgetbridge.service.devices.pebble.PebbleIoThread.connect(PebbleIoThread.java:170)                                                                  
10-29 14:04:46.206 20413 14123 E AndroidRuntime:    at nodomain.freeyourgadget.gadgetbridge.service.devices.pebble.PebbleIoThread.run(PebbleIoThread.java:197)                                                                      
10-29 14:04:46.216   811  1670 W ActivityManager:   Force finishing activity nodomain.freeyourgadget.gadgetbridge/.activities.ControlCenterv2                                                                                    
10-29 14:04:46.316   811   905 D ISSUE_DEBUG: InputChannelName : 9294e15 Application Error: nodomain.freeyourgadget.gadgetbridge                                                                                                  
```

**Comment by aveao** [comment]
I get same error as @VmKid when connecting with the option enabled.

Disabling it makes the app continue functioning.

**Comment by VmKid** [comment]
Should I open a separate issue for the JS crash?

**Comment by danielegobbetti** [comment]
Oh yes please do! That problem is still there. Thanks!

**Comment by VmKid** [comment]
No problem. Cheers.
